<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OpenCV_ConcludeClass</class>
 <widget class="QMainWindow" name="OpenCV_ConcludeClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1132</width>
    <height>1052</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1132</width>
    <height>1052</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1132</width>
    <height>1052</height>
   </size>
  </property>
  <property name="font">
   <font>
    <pointsize>11</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>图像处理工具</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>251</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>选择图像</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>0</y>
      <width>640</width>
      <height>640</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="text">
     <string>这里会显示图像</string>
    </property>
    <property name="scaledContents">
     <bool>false</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QTableWidget" name="modifiedTable">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>640</y>
      <width>341</width>
      <height>411</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>361</height>
     </size>
    </property>
    <property name="font">
     <font>
      <pointsize>11</pointsize>
     </font>
    </property>
    <column>
     <property name="text">
      <string>操作</string>
     </property>
    </column>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>70</y>
      <width>481</width>
      <height>571</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="tabShape">
     <enum>QTabWidget::Triangular</enum>
    </property>
    <property name="currentIndex">
     <number>3</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>二值化</string>
     </attribute>
     <widget class="QPushButton" name="otsu_binary">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>451</width>
        <height>71</height>
       </rect>
      </property>
      <property name="text">
       <string>全局自适应二值化</string>
      </property>
     </widget>
     <widget class="QFrame" name="frame">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>390</y>
        <width>451</width>
        <height>151</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <widget class="QPushButton" name="thresh_binary">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>0</y>
         <width>431</width>
         <height>71</height>
        </rect>
       </property>
       <property name="text">
        <string>阈值二值化</string>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QLineEdit" name="thresh">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>80</y>
         <width>211</width>
         <height>51</height>
        </rect>
       </property>
       <property name="maxLength">
        <number>3</number>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="placeholderText">
        <string>阈值</string>
       </property>
      </widget>
     </widget>
     <widget class="QFrame" name="frame_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>160</y>
        <width>451</width>
        <height>211</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <widget class="QPushButton" name="adaptive_binary">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>0</y>
         <width>431</width>
         <height>71</height>
        </rect>
       </property>
       <property name="text">
        <string>局部自适应二值化</string>
       </property>
      </widget>
      <widget class="QComboBox" name="comboBox">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>80</y>
         <width>191</width>
         <height>41</height>
        </rect>
       </property>
       <property name="frame">
        <bool>true</bool>
       </property>
       <item>
        <property name="text">
         <string>局部平均值法</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>局部高斯加权法</string>
        </property>
       </item>
      </widget>
      <widget class="QWidget" name="widget" native="true">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>130</y>
         <width>291</width>
         <height>61</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Kernel_Size</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spinBox">
          <property name="minimum">
           <number>3</number>
          </property>
          <property name="maximum">
           <number>99</number>
          </property>
          <property name="singleStep">
           <number>2</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QPushButton" name="toGray">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>221</width>
        <height>71</height>
       </rect>
      </property>
      <property name="text">
       <string>灰度处理(1通道化)</string>
      </property>
     </widget>
     <widget class="QPushButton" name="binary_inv">
      <property name="geometry">
       <rect>
        <x>231</x>
        <y>10</y>
        <width>231</width>
        <height>71</height>
       </rect>
      </property>
      <property name="text">
       <string>黑白背景变化</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>图像模糊</string>
     </attribute>
     <widget class="QFrame" name="frame_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>461</width>
        <height>141</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <widget class="QPushButton" name="median_btn">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>441</width>
         <height>71</height>
        </rect>
       </property>
       <property name="text">
        <string>中值滤波</string>
       </property>
      </widget>
      <widget class="QWidget" name="layoutWidget_2">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>90</y>
         <width>221</width>
         <height>41</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Kernel_Size:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="median_kernel_size">
          <property name="minimum">
           <number>3</number>
          </property>
          <property name="singleStep">
           <number>2</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QFrame" name="frame_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>170</y>
        <width>461</width>
        <height>141</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <widget class="QPushButton" name="mean_btn">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>441</width>
         <height>71</height>
        </rect>
       </property>
       <property name="text">
        <string>均值滤波</string>
       </property>
      </widget>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>90</y>
         <width>401</width>
         <height>41</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Kernel_Size:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="mean_x">
          <property name="minimum">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>X</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="mean_y">
          <property name="minimum">
           <number>1</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QFrame" name="frame_5">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>350</y>
        <width>451</width>
        <height>171</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <widget class="QWidget" name="layoutWidget_3">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>100</y>
         <width>401</width>
         <height>41</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Kernel_Size:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="gaussian_x">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="singleStep">
           <number>2</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>X</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="gaussian_y">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="singleStep">
           <number>2</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QPushButton" name="gaussian_btn">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>431</width>
         <height>71</height>
        </rect>
       </property>
       <property name="text">
        <string>高斯滤波</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_4">
     <attribute name="title">
      <string>形态学</string>
     </attribute>
     <widget class="QFrame" name="frame_6">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>50</y>
        <width>471</width>
        <height>491</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <widget class="QPushButton" name="close_btn">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>220</y>
         <width>451</width>
         <height>56</height>
        </rect>
       </property>
       <property name="text">
        <string>闭操作</string>
       </property>
      </widget>
      <widget class="QPushButton" name="dilate_btn">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>451</width>
         <height>56</height>
        </rect>
       </property>
       <property name="text">
        <string>膨胀</string>
       </property>
      </widget>
      <widget class="QPushButton" name="open_btn">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>150</y>
         <width>451</width>
         <height>56</height>
        </rect>
       </property>
       <property name="text">
        <string>开操作</string>
       </property>
      </widget>
      <widget class="QPushButton" name="erode_btn">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>451</width>
         <height>56</height>
        </rect>
       </property>
       <property name="text">
        <string>腐蚀</string>
       </property>
      </widget>
      <widget class="QPushButton" name="tophat_btn">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>290</y>
         <width>451</width>
         <height>56</height>
        </rect>
       </property>
       <property name="text">
        <string>顶帽</string>
       </property>
      </widget>
      <widget class="QPushButton" name="blackhat_btn">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>360</y>
         <width>451</width>
         <height>56</height>
        </rect>
       </property>
       <property name="text">
        <string>黑帽</string>
       </property>
      </widget>
      <widget class="QPushButton" name="gradient_btn">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>430</y>
         <width>451</width>
         <height>56</height>
        </rect>
       </property>
       <property name="text">
        <string>梯度</string>
       </property>
      </widget>
     </widget>
     <widget class="QFrame" name="frame_7">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>0</y>
        <width>341</width>
        <height>51</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Kernel_Size:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="morph_kernel_x">
         <property name="minimum">
          <number>1</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>X</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="morph_kernel_y">
         <property name="minimum">
          <number>1</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>图像边缘</string>
     </attribute>
     <widget class="QFrame" name="frame_8">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>90</y>
        <width>471</width>
        <height>80</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <widget class="QPushButton" name="sobel_y_btn">
       <property name="geometry">
        <rect>
         <x>320</x>
         <y>10</y>
         <width>141</width>
         <height>61</height>
        </rect>
       </property>
       <property name="text">
        <string>Sobel算子_Y</string>
       </property>
      </widget>
      <widget class="QPushButton" name="sobel_btn">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>10</y>
         <width>141</width>
         <height>61</height>
        </rect>
       </property>
       <property name="text">
        <string>Sobel算子</string>
       </property>
      </widget>
      <widget class="QPushButton" name="sobel_x_btn">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>10</y>
         <width>141</width>
         <height>61</height>
        </rect>
       </property>
       <property name="text">
        <string>Sobel算子_X</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>10</y>
        <width>141</width>
        <height>71</height>
       </rect>
      </property>
      <property name="title">
       <string>算子核大小</string>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <widget class="QRadioButton" name="radioButton">
         <property name="text">
          <string>3</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radioButton_2">
         <property name="text">
          <string>5</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QPushButton" name="laplacian_btn">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>290</y>
        <width>441</width>
        <height>61</height>
       </rect>
      </property>
      <property name="text">
       <string>Laplacian算子</string>
      </property>
     </widget>
     <widget class="QFrame" name="frame_9">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>190</y>
        <width>471</width>
        <height>80</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <widget class="QPushButton" name="scharr_y_btn">
       <property name="geometry">
        <rect>
         <x>320</x>
         <y>10</y>
         <width>141</width>
         <height>61</height>
        </rect>
       </property>
       <property name="text">
        <string>Scharr算子_Y</string>
       </property>
      </widget>
      <widget class="QPushButton" name="scharr_btn">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>10</y>
         <width>141</width>
         <height>61</height>
        </rect>
       </property>
       <property name="text">
        <string>Scharr算子</string>
       </property>
      </widget>
      <widget class="QPushButton" name="scharr_x_btn">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>10</y>
         <width>141</width>
         <height>61</height>
        </rect>
       </property>
       <property name="text">
        <string>Scharr算子_X</string>
       </property>
      </widget>
     </widget>
     <widget class="QFrame" name="frame_10">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>370</y>
        <width>451</width>
        <height>161</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <widget class="QPushButton" name="canny_btn">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>431</width>
         <height>61</height>
        </rect>
       </property>
       <property name="text">
        <string>Canny算法</string>
       </property>
      </widget>
      <widget class="QComboBox" name="comboBox_2">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>90</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="currentIndex">
        <number>1</number>
       </property>
       <item>
        <property name="text">
         <string>弱</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>中</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>强</string>
        </property>
       </item>
      </widget>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>90</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>强度:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QPushButton" name="save_img">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>0</y>
      <width>241</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>保存图像</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget_2">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>650</y>
      <width>791</width>
      <height>401</height>
     </rect>
    </property>
    <property name="tabPosition">
     <enum>QTabWidget::North</enum>
    </property>
    <property name="tabShape">
     <enum>QTabWidget::Triangular</enum>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <property name="elideMode">
     <enum>Qt::ElideNone</enum>
    </property>
    <widget class="QWidget" name="tab_5">
     <attribute name="title">
      <string>待写操作_1</string>
     </attribute>
    </widget>
    <widget class="QWidget" name="tab_6">
     <attribute name="title">
      <string>待写操作_2</string>
     </attribute>
    </widget>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="OpenCV_Conclude.qrc"/>
 </resources>
 <connections/>
</ui>
